File: Rubyとか/08_Leader_Dominator.rb
# you can write to stdout for debugging purposes, e.g.
# puts "this is a debug message"

def solution(a)
    # write your code in Ruby 2.2
    hash = Hash.new(0)
    a.each do |aa|
      hash[aa] += 1
    end
  
    half = a.length / 2
    hash.each_key do |key|
      if hash[key] > half 
        f = a.index{|aa| aa == key}
        #puts "hash:#{hash[key]} key:#{key} find:#{f}"
        return a.index{|aa| aa == key}
      end  
    end
    return -1
end
  

File: Python/08_Leader_Dominator.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")

def solution(A):
    # write your code in Python 3.6
    dic = {}
    for a in A:
        if a in dic: dic[a] += 1
        else: dic[a] = 1
    # end of for

    items = dic.items() 
    items = sorted(items, key=lambda x: x[1], reverse=True)
    #print(items)
    if len(items) == 0 or items[0][1] <= len(A) / 2: return -1 
    else: return A.index(items[0][0])
# end of func